src/x86_64/region.h

```diff
@@ -26,6 +26,7 @@ static inline region create_region(u64 base, u64 length, int type)
     r->base = base;
     r->length = length;
     r->type = type;
+    (--r)->type = 0;
```

Comment:
Will actually return the next region...but I guess nothing uses the return val?

Reply:
Oops. Fixed!